creator-node/src/snapbackSM/secondarySyncHealthTracker.js

```diff
@@ -0,0 +1,149 @@
+      const pattern = SecondarySyncHealthTracker._getRedisKeyPattern('*', '*', '*', '*', '*')
+
+      return SecondarySyncHealthTracker._getMetricsMatchingPattern(pattern)
+    } catch (e) {
+      genericLogger.error(`secondarySyncHealthTracker - getAllSyncMetrics() Error || ${e.message}`)
+      return {}
+    }
+  },
+
+  /**
+   * Get SyncRequest outcome metrics for given secondary, all users, and all days
+   */
+  async getSyncMetricsForSecondary (secondary) {
+    try {
+      const pattern = SecondarySyncHealthTracker._getRedisKeyPattern(secondary, '*', '*', '*', '*')
+
+      return SecondarySyncHealthTracker._getMetricsMatchingPattern(pattern)
+    } catch (e) {
+      genericLogger.error(`secondarySyncHealthTracker - getSyncMetricsForSecondary() Error || ${e.message}`)
+      return {}
+    }
+  },
+
+  /**
+   * Get SyncRequest outcome metrics for given secondary, given user, and all days
+   */
+  async getSecondarySyncMetricsForSecondaryForUser (secondary, wallet) {
+    try {
+      const pattern = SecondarySyncHealthTracker._getRedisKeyPattern(secondary, wallet, '*', '*', '*')
+
+      return SecondarySyncHealthTracker._getMetricsMatchingPattern(pattern)
+    } catch (e) {
+      genericLogger.error(`secondarySyncHealthTracker - getSecondarySyncMetricsForUser() Error || ${e.message}`)
+      return {}
+    }
+  },
+
+  /**
+   * Get SyncRequest outcome metrics for given secondary, given user, and for today
+   */
+  async getSecondarySyncMetricsForSecondaryForUserForToday (secondary, wallet) {
+    try {
+      const today = new Date().toISOString().split('T')[0] // format: YYYY-MM-DD
+      const pattern = SecondarySyncHealthTracker._getRedisKeyPattern(secondary, wallet, '*', '*', today)
+
+      return SecondarySyncHealthTracker._getMetricsMatchingPattern(pattern)
+    } catch (e) {
+      genericLogger.error(`secondarySyncHealthTracker - getSecondarySyncMetricsForUserForToday() Error || ${e.message}`)
+      return {}
+    }
+  },
+
+  async _getMetricsMatchingPattern (pattern) {
+    const keys = await SecondarySyncHealthTracker._getAllKeysMatchingPattern(pattern)
+
+    if (!keys || !keys.length) {
+      return {}
+    }
+
+    // This works because vals.length === keys.length
+    const vals = await redisClient.mget(keys)
+
+    // Zip keys and vals arrays into map of key-val pairs
+    const keyMap = {}
+    for (let i = 0; i < keys.length; i++) {
+      keyMap[keys[i]] = vals[i]
+    }
+
+    return keyMap
+  },
+
+  /**
+   * Returns array of all keys in Redis matching pattern, using redis SCAN
+   */
+  async _getAllKeysMatchingPattern (pattern) {
+    const stream = redisClient.scanStream({ match: pattern })
+
+    let keySet = new Set()
+    return new Promise((resolve, reject) => {
+      stream.on('data', async (keys = []) => {
+        keys.forEach(key => { keySet.add(key) })
+      })
+      stream.on('end', () => {
+        resolve(Array.from(keySet).filter(Boolean))
+      })
+      // stream.on('error'),
```

Comment:
i think we should have an error condition here. looks like it can emit error https://github.com/luin/ioredis/blob/master/lib/ScanStream.ts

Reply:
good catch fixed